website/docs/cdktf/concepts/resources.mdx

```diff
@@ -381,6 +381,240 @@ If you need to use the special [`self` object](/terraform/language/resources/pro
+             ~ resource "aws_s3_bucket" "bucket" {
+                   arn                         = "arn:aws:s3:::best-bucket-in-the-world"
+                   bucket                      = "best-bucket-in-the-world"
+                   bucket_domain_name          = "best-bucket-in-the-world.s3.amazonaws.com"
+                   bucket_regional_domain_name = "best-bucket-in-the-world.s3.us-east-1.amazonaws.com"
+                 + force_destroy               = false
+                   hosted_zone_id              = "XXXXXXXXXXXXX"
+                   id                          = "best-bucket-in-the-world"
+                   object_lock_enabled         = false
+                   region                      = "us-east-1"
+                   request_payer               = "BucketOwner"
+                 ~ tags                        = {
+                     - "foo" = "bar" -> null
+                   }
+                 ~ tags_all                    = {
+                     - "foo" = "bar"
+                   } -> (known after apply)
+
+                   grant {
+                       id          = "XXXXXXXXXXXXX"
+                       permissions = [
+                           "FULL_CONTROL",
+                       ]
+                       type        = "CanonicalUser"
+                   }
+
+                   server_side_encryption_configuration {
+                       rule {
+                           bucket_key_enabled = true
+
+                           apply_server_side_encryption_by_default {
+                               sse_algorithm = "AES256"
+                           }
+                       }
+                   }
+
+                   versioning {
+                       enabled    = true
+                       mfa_delete = false
+                   }
+               }
+
+           Plan: 1 to import, 0 to add, 1 to change, 0 to destroy.
+
+           ─────────────────────────────────────────────────────────────────────────────
+
+           Saved the plan to: plan
+
+           To perform exactly these actions, run the following command to apply:
+               terraform apply "plan"
+```
+
+### Generate Configuration For Import
+
+If you want to specify the configuration of your imported resource you can use the static method `generateConfigForImport` on the class of the resource you want to import. This method takes the scope as the first argument, the resource ID of the resource to import to (as will be given in the generated config returned), the resource ID of the resource to be imported, and the provider as an optional fourth. The provider is only required if you have multiple providers of the same type in your configuration.
+
+```typescript
+S3Bucket.generateConfigForImport(this, "bucket", bucketId);
+```
+
+When running `cdktf plan <stack-name>` you will get the information that resource(s) is/are going to be imported, followed by the configuration of your import(s) in CDKTF specific terms.
```

Comment:
I found the original sentence to be a bit too abstract, dunno if this is better though:
```suggestion
When running `cdktf plan <stack-name>` Terraform will generate code for the resource you are importing and CDKTF will convert it to the language you are using.
```